README.md

```diff
@@ -33,7 +33,80 @@ We welcome contributions to OSIPI! To contribute, follow these steps:
 - Push your changes to your fork.
 - Submit a pull request to the main OSIPI repository.
 
-For more details on how to contribute, visit the [Developer Guide](https://osipi.github.io/pypi/developers_guide/index.html).
+## Development Setup
+
+There are two ways to set up your development environment for this project. You can choose the one that suits you best.
+
+### Option 1: Using venv and requirements.txt
+
+1. Clone the repository to your local machine.
+2. Navigate to the project directory.
+3. Create a virtual environment using `venv` by running the following command:
+
+    ```bash
+    python -m venv venv
+    ```
+
+4. Activate the virtual environment:
+
+    - On Windows, run:
+
+        ```bash
+        .\venv\Scripts\activate
+        ```
+
+    - On Unix or MacOS, run:
+
+        ```bash
+        source venv/bin/activate
+        ```
+
+5. Install the required packages from `requirements.txt`:
+
+    ```bash
+    pip install -r requirements.txt
+    ```
+
+### Option 2: Using Poetry
```

Comment:
We want to make poetry a requirement. It's small overhead and it ensures packages won't be missing/incompatible. Remove this and the associated requirements.txt please.